rest-client-base/src/main/java/com/wultra/core/rest/client/base/DefaultRestClient.java

```diff
@@ -0,0 +1,700 @@
+    }
+
+    /**
+     * Private constructor for builder.
+     * @param builder REST client builder.
+     * @throws RestClientException Thrown in case client initialization fails.
+     */
+    private DefaultRestClient(Builder builder) throws RestClientException {
+        this.config = builder.config;
+        // Use WebClient settings from the builder
+        initializeWebClient();
+    }
+
+    /**
+     * Initialize WebClient instance and configure it based on client configuration.
+     */
+    private void initializeWebClient() throws RestClientException {
+        if (config.getBaseUrl() != null) {
+            try {
+                new URI(config.getBaseUrl());
+            } catch (URISyntaxException ex) {
+                throw new RestClientException("Invalid parameter baseUrl");
+            }
+        }
+        WebClient.Builder builder = WebClient.builder();
+        HttpClient httpClient = HttpClient.create();
+        SslContext sslContext;
+        try {
+            if (config.isAcceptInvalidSslCertificate()) {
+                sslContext = SslContextBuilder
+                        .forClient()
+                        .trustManager(InsecureTrustManagerFactory.INSTANCE)
+                        .build();
+                httpClient = httpClient.secure(sslContextSpec -> sslContextSpec.sslContext(sslContext));
+            }
+        } catch (SSLException ex) {
+            throw new RestClientException("SSL error occurred: " + ex.getMessage(), ex);
+        }
+        httpClient = httpClient.tcpConfiguration(tcpClient -> {
+                            if (config.getConnectionTimeout() != null) {
+                                tcpClient = tcpClient.option(
+                                        ChannelOption.CONNECT_TIMEOUT_MILLIS,
+                                        config.getConnectionTimeout());
+                            }
+                            if (config.isProxyEnabled()) {
+                                tcpClient = tcpClient.proxy(proxySpec -> {
+                                    ProxyProvider.Builder proxyBuilder = proxySpec
+                                            .type(ProxyProvider.Proxy.HTTP)
+                                            .host(config.getProxyHost())
+                                            .port(config.getProxyPort());
+                                    if (config.getProxyUsername() != null && !config.getProxyUsername().isEmpty()) {
+                                        proxyBuilder.username(config.getProxyUsername());
+                                        proxyBuilder.password(s -> config.getProxyPassword());
+                                    }
+                                    proxyBuilder.build();
+                                });
+                            }
+                            return tcpClient;
+                        }
+                );
+        builder.exchangeStrategies(ExchangeStrategies.builder()
```

Comment:
I would extract the `ExchangeStrategies.builder()...build()` call in front of `builder.exchangeStrategies(...)` call, to make the code a bit more readable.

Reply:
Ok, fixed.